_posts/2019-10-26-dividing-human-pie-ai.md

```diff
@@ -0,0 +1,59 @@
+
+But given the participants' diversity of experience, how should we divide the groups to give everyone a better experience?
+
+There were so many possible ways to divide our human pie that there was no clear answer. I needed some data!
+
+During a morning break I asked all of the participants to rate themselves in terms of their machine learning knowledge, where on the developer - data scientist spectrum they identify themselves, and how much cloud experience they had, using a Google Form.
+
+I fed the responses into a Python app that used the [_scikit-learn_](https://scikit-learn.org/) library's [_k_-means clustering algorithm](https://scikit-learn.org/stable/modules/clustering.html#k-means) to assign the responses into two groups. 
+
+_k_-means is a relatively simple unsupervised learning algorithm that can be used to group similar data points together. 
+
+It starts by randomly choosing locations for _k_ cluster centroids, where _k_ is the desired number of groups. The distances to the data points are then calculated, and data points are each assigned to their nearest centroid. The algorithm then moves each centroid to reduce the sum of the distances between the data points and their respective centroids.
+
+The process is then repeated until no data points change centroids. At this point the algorithm is said to have 'converged' on a solution.
+
+This process is repeated a number of times with different initial centroid locations to try and ensure a robust solution. It isn't guaranteed to be the optimum solution, but in this case any reasonably good solution would be acceptable. 
+
+## Mmm delicious pie
+The _k_-means algorithm conveniently divided the participants into data scientists and developers. This was a gratifyingly simple solution, and it worked really well in practice. 
+
+My biggest concern was the group sizes would be dramatically unbalanced. Ideally I wanted them to be similarly sized, but the _k_-means clustering algorithm doesn't guarantee this. As luck would have it the algorithm split the participants into 11 data scientists and 14 developers, which was close enough. In the 3D visualisation below the points in red represent the data scientists, and the green points represent the developers.
+
+![A 3D visualisation of our response data]({{ "/images/2019-10-26-3d-visualisation.png"}})
+
+The group of developers first learned about Amazon SageMaker and then went on to build a chatbot in Amazon Lex. The data scientists decided that instead of following the Lex demo they'd rather get stuck into the time series forecasting and product recommendation demos. These required a bit more data science understanding and were more relevant to the projects that they were working on.
```

Comment:
```suggestion
With our large group now split into two smaller 'focus' groups, we could now have a 'data scientists' group and a 'developers' group.  The group of developers first learned about Amazon SageMaker and then went on to build a chatbot in Amazon Lex.  The data scientists, however, decided that instead of following the Lex demo they'd rather get stuck into the time series forecasting and product recommendation demos. These required a bit more data science understanding and were more relevant to the projects that they were working on.
```

Reply:
Good suggestion, sounds better.